sos/report/plugins/openstack_designate.py

```diff
@@ -19,6 +19,17 @@ class OpenStackDesignate(Plugin):
     var_puppet_gen = "/var/lib/config-data/puppet-generated/designate"
 
     def setup(self):
+        # collect current pool config
+        pools_cmd = "designate-manage pool generate_file --file /dev/stdout"
+
+        if self.container_exists(".*designate_central"):
+            pools_cmd = self.fmt_container_cmd("designate_central", pools_cmd)
```

Comment:
Has this been tested on a real OSP environment? Is the container name really, always, just `designate_central`? If so we can drop the `.*` in the `container_exists()` call - otherwise we'll want to do something like:

```python
pools_cmd = self.fmt_container_cmd(self.get_container_by_name('.*designate_central'), pools_cmd)
```

Also, `fmt_container_cmd()` already does a `container_exists()` check internally, so we can drop line 25. If the container doesn't exist, it just returns the same `cmd` that you hand to it.

Reply:
Fixed, thanks.

Reply:
Sorry, let me clarify a bit:

`self.container_exists()` can take a regex and simply checks if a given string matches _any_ container's name.
`self.fmt_container_cmd()` does _not_ take a regex, but must take an explicit container name. It does a check to make sure a container with that exact name actually exists before doing any formatting of the command string.
`self.get_container_by_name()` takes a regex and returns the first matching container.

The key here is that if the container name is not truly static, e.g. `designate_central`, but does have some predictable element, e.g. `qwerty-designate_central`, then you'd use `get_container_by_name()` to retrieve the actual name of the container to use with `fmt_container_cmd()`.

Reply:
My bad, read your previous reply too quickly. Should be fixed now.